src/hupper/ipc.py

```diff
@@ -157,16 +161,23 @@ def _recv_packet(self):
             remaining -= n
         return pickle.loads(buf.getvalue())
 
+    def _on_recv(self, packet):
+        if self.on_recv is not None:
+            self.on_recv(packet)
+
     def _read_loop(self):
         try:
             while True:
                 packet = self._recv_packet()
                 if packet is None:
                     break
-                self.on_recv(packet)
+                self._on_recv(packet)
         except EOFError:
             pass
-        self.on_recv(None)
+        except OSError as e:
+            if e.errno != errno.EBADF:
+                raise
+        self._on_recv(None)
```

Comment:
what is the goal here of calling `_on_recv(None)`?

Reply:
I don't know, the code was doing it before